oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -765,9 +777,26 @@ int mca_spml_ucx_fence(shmem_ctx_t ctx)
 
 int mca_spml_ucx_quiet(shmem_ctx_t ctx)
 {
+    int flush_get_data;
     int ret;
+    size_t i, nprocs;
     mca_spml_ucx_ctx_t *ucx_ctx = (mca_spml_ucx_ctx_t *)ctx;
 
+    if (mca_spml_ucx.synchronized_quiet) {
+        nprocs = oshmem_num_procs();
+        for (i = 0; i < nprocs; i++) {
```

Comment:
so quiet will be o(n)..
maybe we can keep the indices of the peers to flush in an array? to avoid going over all peers

Reply:
👍 You can manage this information on per-context basis (internally in UCX worker).

Reply:
in case if keep indices we have to check for duplicated indices on every put operation which could be more expensive

Reply:
it happens only if sync flush is enabled, but it will make sure we flush only relevant procs
+let's perform the flush in random order to avoid congestion

Reply:
re-implemented using bitmap + array pair